bee-network/bee-gossip/src/swarm/protocols/iota_gossip/io.rs

```diff
@@ -28,89 +28,80 @@ pub fn channel() -> (GossipSender, GossipReceiver) {
     (sender, UnboundedReceiverStream::new(receiver))
 }
 
-pub fn start_incoming_processor(
+pub fn start_inbound_gossip_handler(
     peer_id: PeerId,
-    mut reader: BufReader<ReadHalf<Box<NegotiatedSubstream>>>,
-    incoming_tx: GossipSender,
-    internal_event_sender: InternalEventSender,
+    mut inbound_gossip_rx: BufReader<ReadHalf<Box<NegotiatedSubstream>>>,
+    inbound_gossip_tx: GossipSender,
+    internal_event_tx: InternalEventSender,
 ) {
     tokio::spawn(async move {
-        let mut msg_buf = vec![0u8; MSG_BUFFER_LEN];
+        let mut buf = vec![0u8; MSG_BUFFER_LEN];
 
         loop {
-            if let Some(len) = (&mut reader).read(&mut msg_buf).await.ok().filter(|len| *len > 0) {
-                if incoming_tx.send(msg_buf[..len].to_vec()).is_err() {
-                    debug!("gossip-in: receiver dropped locally.");
+            if let Some(len) = (&mut inbound_gossip_rx)
+                .read(&mut buf)
+                .await
+                .ok()
+                .filter(|len| *len > 0)
+            {
+                if inbound_gossip_tx.send(buf[..len].to_vec()).is_err() {
+                    debug!("Terminating gossip protocol with {}.", alias!(peer_id));
 
-                    // The receiver of this channel was dropped, maybe due to a shutdown. There is nothing we can do
-                    // to salvage this situation, hence we drop the connection.
                     break;
                 }
             } else {
-                debug!("gossip-in: stream closed remotely.");
+                debug!("Peer {} terminated gossip protocol.", alias!(peer_id));
```

Comment:
The nesting here is a little bit difficult to read imo, could it be reorganised?

Reply:
Ah, no worries 👍